src/drt/src/rp/FlexRP_prep.cpp

```diff
@@ -753,6 +755,217 @@ void FlexRP::prep_via2viaForbiddenLen_helper(const frLayerNum& lNum,
   }
 }
 
+bool FlexRP::hasMinStepViol(Rect& r1, Rect& r2, frLayerNum lNum) {
+    gtl::rectangle_data<frCoord> rect1(
+        r1.xMin(), r1.yMin(), r1.xMax(), r1.yMax());
+    gtl::rectangle_data<frCoord> rect2(r2.xMin(),
+                                             r2.yMin(),
+                                             r2.xMax(),
+                                             r2.yMax());
+
+    // joining the two via rects in one polygon
+    gtl::polygon_90_set_data<frCoord> set;
+    using namespace boost::polygon::operators;
+    set += rect1;
+    set += rect2;
+    std::vector<gtl::polygon_90_with_holes_data<frCoord>> polys;
+    set.get_polygons(polys);
+    if (polys.size() != 1)
+      return false;
+
+    gtl::polygon_90_with_holes_data<frCoord> poly = *polys.begin();
+    gcNet* testNet = new gcNet(0);
+    gcPin* testPin = new gcPin(poly, lNum, testNet);
```

Comment:
Are these leaked?  You might want a unique_ptr here.

Reply:
This code was actually copied from the code that Osama created to do DRC for minstep when handling via-via junctions. I will take a look at it.